lib/input/reader/mqtt.go

```diff
@@ -77,13 +82,19 @@ func NewMQTT(
 		log:           log,
 	}
 
+	var err error
 	if len(conf.StaleConnectionTimeout) > 0 {
 		var err error
 		if m.staleConnectionTimeout, err = time.ParseDuration(conf.StaleConnectionTimeout); err != nil {
 			return nil, fmt.Errorf("unable to parse stale connection timeout duration string: %w", err)
 		}
 	}
 
+	err = m.conf.Will.Validate()
```

Comment:
The first `var err error` should be possible to remove and then do `err := ..Validate()` ?

Reply:
The style is good when you expect to add additional error guards to the same function at a later time, since that future diff will not touch the first err guard. This style follow a common pattern in this code base. Ultimately, it is a YAGNI judgement call perhaps, but this style is a common pattern in this code base.